DietPi-Globals | Renamed G_USER_INPUTS to G_INTERACTIVE

alpha-tango-kilo opened the pull request:
<!--
Before submitting a pull request:
- Please ensure the target branch is "dev" (active development): https://github.com/MichaIng/DietPi/tree/dev
- Please ensure changes have been tested and verified functional.
-->
**Status**: Ready
Needs checking that it actually works in the real world (specifically the backwards compatibility with scripts), but exporting a value to `G_USER_INPUTS` is picked up by `dietpi-globals` and `G_INTERACTIVE` is then given the value

**Reference**: https://github.com/MichaIng/DietPi/pull/3033#issuecomment-518043583

**Commit list/description**:
1. I used ~~skill~~ find to replace all instances of `G_USER_INPUTS` with `G_INTERACTIVE`
2. Ignore this commit, aliases don't work in scripts
3. I fixed it, making the variable name change backwards compatible

The point of the PR is basically to make the global variable's name more self-explanatory, while being backwards compatible for the sake of any existing users' scripts

Commit:
Change G_USER_INPUTS to G_INTERACTIVE
Find and replace of 37 instances across the whole repo

Commit:
Alias G_USER_INPUTS to G_INTERACTIVE
If this works, then the PR remains backwards compatible

Commit:
Changed `alias` to not-null testing, works!
https://stackoverflow.com/questions/3601515/how-to-check-if-a-variable-is-set-in-bash

MichaIng commented:
@AtkLordOverAll 
Many thanks for this, looks good.
I will move the `G_USER_INPUTS` handling into the `G_INTERACTIVE` existence check, so the latter has priority and if it exists, a tiny step less is done.

MichaIng committed:
v6.26

+ DietPi-Globals | Apply G_USER_INPUTS backwards compatibility only if G_INTERACTIVE is not set, to make the latter take priority

MichaIng commented on dietpi/func/dietpi-globals:
```diff
@@ -34,10 +34,18 @@
 
-		[[ -t 0 ]] && G_INTERACTIVE=1 || G_INTERACTIVE=0
+		# Backwards compatibility to keep user scripts valid for a while
+		if [[ $G_USER_INPUTS == [01] ]]; then
```
I switched to [01] testing, so in case any invalid value has been given, it is reverted to auto detection via STDIN check.

MichaIng commented:
Changelog: https://github.com/MichaIng/DietPi/commit/03b883c87de031f386e2a189daaebb4d84316a45
